### DISCUSSION TITLE
Which certificate should I buy for signing the exe?

### DISCUSSION THREAD

**Discussion by riccardospeggiorin-centropaghe** [rustdesk/rustdesk#8911] [General]
Hi, quick question. 
I am trying to test the self hosted option...everything works good expect the "ugly" name of the exe for storing the relay server and the private key. I learn that I need to build and sign the exe with a certificate.
Which one should I get?
Thanks

**Comment by rustdesk** [comment]
basic plan

**Comment by riccardospeggiorin-centropaghe** [comment]
I meant the certificate for sign the exe

**Comment by danielmarschall** [comment]
To get a custom client without "ugly filename", you can build it using GitHub. See tutorial here: https://rustdesk.com/docs/en/dev/build/all/

That's all you need. Signing the EXE file is optional.

**_Regarding signing:_**

When people download the EXE file, it might be blocked by Microsoft Smartscreen. To get rid off that, you have a few options:

- A) Send the file to Microsoft to whitelist it https://www.microsoft.com/en-us/wdsi/filesubmission
- B) A lot of people must download the EXE, so it gets a high reputation
- C) Buy an EV certificate (is more expensive and your organization will be verified)

Regarding certificate:

- If you are an Open Source developer, check out Certum Open Source Code Signing. It costs 69 EUR.
- If you are an Organization, you can use any OV Code Signing certificate 
- If you want auto-whitelist to Microsoft, search for EV Code Signing

Personally, I am using a GlobalSign EV certificate, but any other CA works fine. It cost 799 EUR for 3 years. I sign locally, which means I download the EXE file which was built by GitHub, and then I sign it using Microsoft SignTool using the USB Token GlobalSign sent me. Setup the token and all that stuff is very time consuming.
